### PR TITLE
Add inline tabs on Quick Reference page for commands on different systems

### PR DESCRIPTION
**Pull request by lancegoyke**
This is for work on #1196

This PR introduces a new dependency: [sphinx-inline-tabs](https://sphinx-inline-tabs.readthedocs.io/en/latest/)

There were two design decisions I am not set on:

1. Should we say "Unix" or "Linux"? I picked "Unix" because the [setup docs](https://devguide.python.org/getting-started/setup-building/#mac-python-exe) say "Unix".
2. Is the most useful order Unix, then macOS, then Windows. I would presume most developers are using macOS, but other examples I've seen start with "Linux" (e.g., [PyPI](https://pip.pypa.io/en/stable/installation/), [Pillow](https://pillow.readthedocs.io/en/stable/installation.html#basic-installation)), so I put "Unix" first.

----
:books: Documentation preview :books:: https://cpython-devguide--1205.org.readthedocs.build/

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
     'sphinx_copybutton',
     'sphinxext.opengraph',
     'sphinxext.rediraffe',
+    'sphinx_inline_tabs',
```
Comment (hugovk) [inline conf.py:16]: Let's sort this list in alphabetical order.

File: index.rst [hunk 2]
```diff
@@ -57,17 +84,25 @@ instructions please see the :ref:`setup guide <setup>`.
...
 
-6. Once you fixed the issue, run the tests, and the patchcheck.
+6. Once you fixed the issue, run the tests, and the patchcheck:
+
+   .. tab:: Unix
+
+      .. code-block:: shell
+
+         make patchcheck
+
```
Comment (hugovk) [inline index.rst:94]: To make the linter happy:
```suggestion

```

**Review by hugovk — COMMENTED** [review]
It's working well at https://cpython-devguide--1205.org.readthedocs.build/ :)

> 1. Should we say "Unix" or "Linux"? I picked "Unix" because the [setup docs](https://devguide.python.org/getting-started/setup-building/#mac-python-exe) say "Unix".

Let's keep the existing terminology for this PR, so "Unix".

> 2. Is the most useful order Unix, then macOS, then Windows. I would presume most developers are using macOS, but other examples I've seen start with "Linux" (e.g., [PyPI](https://pip.pypa.io/en/stable/installation/), [Pillow](https://pillow.readthedocs.io/en/stable/installation.html#basic-installation)), so I put "Unix" first.


Without knowing what operating systems our visitors use, I'd probably stick with the same order as given in the text: Unix, macOS, then Windows.

(For the main docs, it's something like 59% Windows, 22% macOS and 10% Linux, but the audience of this site could be different.)

**Comment by Mariatta** [comment]
I looked at the preview and it looks great. Thanks for working on this.

File: index.rst [hunk 3]
```diff
@@ -27,27 +27,54 @@ instructions please see the :ref:`setup guide <setup>`.
       git clone https://github.com/<your_username>/cpython
       cd cpython
 
-3. Build Python, on Unix and macOS use::
+3. Build Python, on Unix and macOS use:
```
Comment (Mariatta) [inline index.rst:30]:
```suggestion
3. Build Python:
```

The tab now shows instructions for Unix, macOS, and Windows.

File: index.rst [hunk 4]
```diff
@@ -27,27 +27,54 @@ instructions please see the :ref:`setup guide <setup>`.
...
+
+   .. tab:: Windows
+
+      .. code-block:: dosbatch
+
+         .\python.bat -m test -j3
+
+   Note: :ref:`Most <mac-python.exe>` macOS systems use
+   :file:`./python.exe` in order to avoid filename conflicts with
+   the ``Python`` directory.
```
Comment (hugovk) [inline index.rst:77]: Shall we put this note inside the macOS tab?

Comment (lancegoyke) [reply index.rst:77]: Yes!

File: index.rst (hunk 3 above)
Comment (hugovk) [reply index.rst:30]: This bit still needs doing.

**Comment by lancegoyke** [comment]
@Mariatta and @hugovk -- thank you for your input! I have updated every item you both raised. 

I made one additional change: `requirements.txt` is now in alphabetical order as that seemed similar to the `extensions` order in `conf.py`.

File: index.rst (hunk 3 above)
Comment (lancegoyke) [reply index.rst:30]: Weird, I know I accepted that suggestion. Updating...

**Review by hugovk — APPROVED** [review]
Thank you!

**Comment by pradyunsg** [comment]
Thanks @lancegoyke! ^.^

**Pull request merged**